### PR TITLE
Addressing .net sdk comments

### LINKED ISSUE #27876: [RiskIQ - EASM - Defender EASM] .NET SDK review follow up
- We had a SDK review meeting last week, and there are follow-up comments that we need to address by updating the TypeSpec specification file
- This is the API view for the .NET sdk: https://apiview.dev/Assemblies/Review/7015babaca8d4076bf0a39a40b04c298#Azure.Analytics.Defender.Easm.InventoryAsset. This should also include the comments from the reviewer

### PR DESCRIPTION
**Pull request by thang-bit**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
- Update .NET projected name for single-word models like Attribute or Banner
- Update .NET projected name Task to TaskResource
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

File: specification/riskiq/Easm/models.tsp [hunk 1]
```diff
@@ -1351,6 +1366,7 @@ enum ReportBillableAssetBreakdownKind {
 
 #suppress "@azure-tools/typespec-azure-core/documentation-required" "Documentation will be added next preview version"
 @doc("The state the task is in.")
+@projectedName("csharp", "TaskResourceState")
```
Comment (timotheeguerin) [inline specification/riskiq/Easm/models.tsp:1369]: You'll want to use `@clientName` in the `client.tsp` here . `@projectedName` is in the process of being deprecated and we are changing all instances

Comment (timotheeguerin) [reply specification/riskiq/Easm/models.tsp:1369]: (same for all other `@projectedName`)

Comment (thang-bit) [reply specification/riskiq/Easm/models.tsp:1369]: @timotheeguerin I'm getting this error when running npx tsv command: 
image
I'm using clientName like this: 
image
Is there any way that I can fix without changing the name of model?

Comment (timotheeguerin) [reply specification/riskiq/Easm/models.tsp:1369]: Also I think they want to keep tcgc decorator in the client.tsp with augment.

```
@@clientName(Source, "SourceDetails", "csharp")
```

Comment (timotheeguerin) [reply specification/riskiq/Easm/models.tsp]: Typespec client generator core

**Comment by thang-bit** [comment]
@timotheeguerin after I use @@clientName instead of projectedName, the generated sdk doesn't rename model as expected. For example I changed model name from DiscoSource to DiscoverySource in .NET. Do you think this is a bug for typespec team?
image
image

**Comment by timotheeguerin** [comment]
> @timotheeguerin after I use @@clientName instead of projectedName, the generated sdk doesn't rename model as expected. For example I changed model name from DiscoSource to DiscoverySource in .NET. Do you think this is a bug for typespec team? image image

is it just changing the name partially?

**Comment by thang-bit** [comment]
> > @timotheeguerin after I use @@clientName instead of projectedName, the generated sdk doesn't rename model as expected. For example I changed model name from DiscoSource to DiscoverySource in .NET. Do you think this is a bug for typespec team? image image
> 
> is it just changing the name partially(in the function types but not the actual model)?

It doesn't change any thing, either in function type or actual model. It happens to all other models and operations

**Review by timotheeguerin — APPROVED** [review]
Looks good on the `@projectedName` change part

**Comment by thang-bit** [comment]
Hi @timotheeguerin, I don't have permission to merge this PR. Do you have the permission to merge this PR?

**Comment by timotheeguerin** [comment]
> Hi @timotheeguerin, I don't have permission to merge this PR. Do you have the permission to merge this PR?

I do not, I am not a spec repo reviewer

**Pull request merged**